library/src/main/java/com/google/android/exoplayer/util/PlayerControl.java

```diff
@@ -97,9 +103,24 @@ public void pause() {
 
   @Override
   public void seekTo(int timeMillis) {
-    long seekPosition = exoPlayer.getDuration() == ExoPlayer.UNKNOWN_TIME ? 0
-        : Math.min(Math.max(0, timeMillis), getDuration());
+    long seekPosition = (long) timeMillis;
+    if (exoPlayer.getDuration() == ExoPlayer.UNKNOWN_TIME) {
+      if (availableRange != null) {
```

Comment:
I had tested with a negative playback position, and it works fine.Please try it yourself and let me know if you find any issue. For testing, I just hardcoded the seekPosition here to some negative value.